src/test/java/api/loans/CheckOutByBarcodeTests.java

```diff
@@ -256,6 +256,22 @@ void canCheckOutUsingItemAndUserBarcode() {
     assertThat(callNumberComponents.getString("suffix"), is("SUFFIX"));
   }
 
+  @Test
+  void checkOutFailedWhenPatronGroupIdIsMissingInUser() {
+    final IndividualResource smallAngryPlanet = itemsFixture.basedUponSmallAngryPlanet();
+    final IndividualResource noUserGroupBob = usersFixture.noUserGroupBob();
+
+    final Response response = checkOutFixture.attemptCheckOutByBarcode(500,
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(smallAngryPlanet)
+        .to(noUserGroupBob)
+        .on(getZonedDateTime())
+        .at(UUID.randomUUID()));
+
+    assertThat(response.getStatusCode(), is(500));
+    assertThat(response.getBody(), is("Unable to apply circulation rules to a user with null value as patronGroupId"));
+  }
```

Comment:
I think the changes should be tested by triggering notice-session-expiration-by-timeout endpoint because it was the main idea of the issue described in CIRC-1286

Reply:
Yes, I also tried to reproduce the bug in EndExpiredPatronActionSessionTests class, but unfortunately, we are not able to create an Item without the required fields. I only could reproduce without patronGroupId in the user. Because patronGroupId is not a required field. 
FYI, I reproduced the bug manually deleting from DB, it is not possible with existing API

Reply:
Is it possible to update the item before triggering notice-session-expiration-by-timeout timer?